book/analysis/performance-of-python-types.md

```diff
@@ -108,36 +92,29 @@ concatenate | $$O(k)$$
 sort    | $$O(n log n)$$
 multiply    | $$O(nk)$$
 
-
 Dictionaries
 ---
 
-The second major Python data structure is the dictionary. As you
-probably recall, dictionaries differ from lists in that you can access
-items in a dictionary by a key rather than a position. Later in this
-book you will see that there are many ways to implement a dictionary.
-The thing that is most important to notice right now is that the get
-item and set item operations on a dictionary are $$O(1)$$. It is
-difficult to provide an intuitive explanation for this without diving
-into the implementation as we do later, except to say that a dictionary
-is a structure created specifically for the purpose of setting and
-retrieving values by key in constant time.
-
-Another important dictionary operation is the contains operation.
-Checking to see whether a key is in the dictionary or not is also $$O(1)$$.
-This follows from the fact that getting an item from a dictionary is
-itself $$O(1)$$.
-
-Iterating over a dictionary is unsurprisingly $$O(n)$$, as is copying
-the dictionary, as $$n$$ individual key/value pairs must be copied.
-
-The efficiency of all dictionary operations is summarized in the table
-below. One important side note on dictionary performance is that the
-efficiencies we provide in the table are for average performance. In
-some rare cases the contains, get item, and set item operations can
-degenerate into $$O(n)$$ performance but we will get into that in a
-later section when we talk about the different ways that a dictionary
-could be implemented.
+The second major Python data type is the dictionary. As you might recall,
+dictionaries differ from lists in their ability to access items by key rather
+than position. For now, the most important characteristic to note is that
+"getting" and "setting" an item in a dictionary are both $$O(1)$$ operations.
+
+We won't immediately attempt to provide an intuitive explanation for this, but
+rest assured that we will dive into a discussion of dictionary implementations
+later. For now, simply remember that dictionaries were created specifically for
+the purpose of getting and setting values by key in constant time.
```

Comment:
Maybe we shouldn't say "in constant time" since not all dictionary implementations would necessarily have strictly constant time search and set. CPython uses hashmaps for dictionaries, but it could well have used trees like Haskell did for its maps. So maybe "as fast as possible"?